whatslab/src/components/Inputs/InputseAreaDeMensagens.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { ContainerGeral,
+         ContainerMensagensPostadas,
+         ContainerMensagemEnviada,
+         ContainerInputs,
+         Negrito } from './Style'
+
+export default class InputseAreaDeMensagens extends React.Component {
+    
+ // estado para os inputs
+  state = { 
+   // array vazio pois o inicio não deve conter nada na tela
+   arrayDadosDoInput: [],
+
+  // variaveis para os dados digitados nos inputs
+   valorInputUsuario: "",
+   valorInputMensagem: ""
+  }
+
+  onClickadicionaPost = (event) => {
+    // O comportamento padrão do Form é atualizar a pagina depois do envio dos dados. O preventDefault previne essa atualização padrão.
+    event.preventDefault()
+
+    // variavel para representar os dados digitados no input
+    const novoPost = { 
+      nome: this.state.valorInputUsuario,
+      mensagem: this.state.valorInputMensagem
+    }
+
+    // Um novo array é criado para poder adicionar os novos posts usando como copia o principal
+    const novoArrayDeDados = [...this.state.arrayDadosDoInput, novoPost] 
+
+    // Agora o array do estado é o novoArrayDeDados criado acima
+    this.setState({ arrayDadosDoInput: novoArrayDeDados }) 
+
+    // para apagar os dados dos inputs pega-se as variaveis dos inputs e coloca-se uma string vazia
+    this.setState({ valorInputMensagem: "", valorInputUsuario: ""})
```

Comment:
Aqui, vcs podem usar apenas um this.seState() para mudar todas as propriedades do estado de uma vez só. Otimiza o código e deixa ele mais limpo:

this.setState({ 
  arrayDadosDoInput: novoArrayDeDados,
  valorInputMensagem: "",
  valorInputUsuario: ""
})